packages/manager/src/features/CloudPulse/shared/CloudPulseRegionSelect.tsx

```diff
@@ -40,13 +56,33 @@ export const CloudPulseRegionSelect = React.memo(
       // eslint-disable-next-line react-hooks/exhaustive-deps
     }, [regions]);
 
+    // validate launchDrakly region_ids with the ids from the fetched 'all-regions'
+    const supportedRegions = React.useMemo<Region[] | undefined>(() => {
+      const serviceTypeFlag = flags.aclpServiceTypeMap?.find(
+        (item) => item.serviceType === selectedDashboard?.service_type
```

Comment:
serviceType and supportedRegionIds are not strongly typed, can we add proper typescript types for type safety as per checklist?